pkgs/applications/office/libreoffice/default.nix

```diff
@@ -557,20 +564,56 @@ in
     zip
     zlib
   ]
-  ++ (with gst_all_1; [
-    gst-libav
-    gst-plugins-bad
-    gst-plugins-base
-    gst-plugins-good
-    gst-plugins-ugly
-    gstreamer
-  ])
+  ++ passthru.gst_packages
   ++ optionals kdeIntegration [ qtbase qtx11extras kcoreaddons kio ]
   ++ optionals (lib.versionAtLeast (lib.versions.majorMinor version) "7.4") [ libwebp ];
 
   passthru = {
     inherit srcs;
     jdk = jre';
+    inherit kdeIntegration;
+    # For the wrapper.nix
+    inherit gtk3;
+    # Although present in qtPackages, we need qtbase.qtPluginPrefix and
+    # qtbase.qtQmlPrefix
+    inherit qtbase;
+    gst_packages = with gst_all_1; [
+      gst-libav
+      gst-plugins-bad
+      gst-plugins-base
+      gst-plugins-good
+      gst-plugins-ugly
+      gstreamer
+    ];
+    qmlPackages = [
+      ki18n
+      knotifications
+      qtdeclarative
+      qtquickcontrols
+      qtwayland
+      solid
+      sonnet
+    ];
+    qtPackages = [
+      kauth
+      kcompletion
+      kconfigwidgets
+      kglobalaccel
+      ki18n
+      kio
+      kitemviews
+      ktextwidgets
+      kwidgetsaddons
+      kwindowsystem
+      kxmlgui
+      phonon
+      qtbase
+      qtdeclarative
+      qtsvg
+      qttools
+      qtwayland
+      sonnet
+    ];
```

Comment:
Why are passing through packages exactly which are not even used?

Reply:
What do you mean not used? These are the packages used for generating the paths in the wrappers.

Reply:
> Why are passing through packages exactly which are not even used?

It's a good question. I took all of the references I found in the original wrapped executable of `libreoffice-qt` that came from the `QT_PLUGIN_PATH` and the `QML2_IMPORT_PATH` environment variables. I'm not sure exactly why all of these dependencies came due to `wrapQt`, but I didn't want to brake anything for any `libreoffice-qt` user out there due to this change.

> What do you mean not used? These are the packages used for generating the paths in the wrappers.

I think @SuperSandro2000 meant that they are not used as dependencies.